src/crawl/DeepCrawler/DeepCrawler.js

```diff
@@ -0,0 +1,161 @@
+import * as cheerio from 'cheerio';
+
+import { BaseCrawler } from '../BaseCrawler.js';
+import { SCRAPE_PAGE_PROMPT, SCRAPE_TYPE_PROMPT, ANALYZE_PAGE_PROMPT, GENERATE_SCHEMA_PROMPT } from './prompts.js';
+
+export class DeepCrawler extends BaseCrawler {
+  // TODO(pilvcc): what is `query` and what do we do with it?
+  async *run(url, query, options) {
```

Comment:
query is basically the kind of link to look for, eg. "find all the links to pokemon on this page"